Crash when ordering air-strike near map edge

## System Information
<!-- Information about the operating system, engine version, game mod and package source are mandatory for investigating and fixing crashes. -->
- OpenRA Version: bleed (4961b0943bd2637a0c976afb5ec95893eef1b95f)
- OpenRA Mod: TD

## Exception log
```

Unhandled Exception:
System.ArgumentException: An item with the same key has already been added. Key: a10 622
  at System.Collections.Generic.Dictionary`2[TKey,TValue].TryInsert (TKey key, TValue value, System.Collections.Generic.InsertionBehavior behavior) [0x0015a] in <b2b5eee6b3734e9586bbc3ab32e40bc7>:0
  at System.Collections.Generic.Dictionary`2[TKey,TValue].Add (TKey key, TValue value) [0x00000] in <b2b5eee6b3734e9586bbc3ab32e40bc7>:0
  at OpenRA.Mods.Common.Traits.AirstrikePower+<>c__DisplayClass4_0.<SendAirstrike>b__3 (OpenRA.World w) [0x00120] in /home/ivaylo/code/OpenRA-dragunoff/OpenRA.Mods.Common/Traits/SupportPowers/AirstrikePower.cs:195
  at OpenRA.World.Tick () [0x0019e] in /home/ivaylo/code/OpenRA-dragunoff/OpenRA.Game/World.cs:472
  at OpenRA.Game.InnerLogicTick (OpenRA.Network.OrderManager orderManager) [0x001bc] in /home/ivaylo/code/OpenRA-dragunoff/OpenRA.Game/Game.cs:602
  at OpenRA.Game.LogicTick () [0x0003e] in /home/ivaylo/code/OpenRA-dragunoff/OpenRA.Game/Game.cs:629
  at OpenRA.Game.Loop () [0x000f1] in /home/ivaylo/code/OpenRA-dragunoff/OpenRA.Game/Game.cs:793
  at OpenRA.Game.Run () [0x0003c] in /home/ivaylo/code/OpenRA-dragunoff/OpenRA.Game/Game.cs:834
  at OpenRA.Game.InitializeAndRun (System.String[] args) [0x00010] in /home/ivaylo/code/OpenRA-dragunoff/OpenRA.Game/Game.cs:262
  at OpenRA.Program.Main (System.String[] args) [0x00014] in /home/ivaylo/code/OpenRA-dragunoff/OpenRA.Game/Support/Program.cs:31
[ERROR] FATAL UNHANDLED EXCEPTION: System.ArgumentException: An item with the same key has already been added. Key: a10 622
  at System.Collections.Generic.Dictionary`2[TKey,TValue].TryInsert (TKey key, TValue value, System.Collections.Generic.InsertionBehavior behavior) [0x0015a] in <b2b5eee6b3734e9586bbc3ab32e40bc7>:0
  at System.Collections.Generic.Dictionary`2[TKey,TValue].Add (TKey key, TValue value) [0x00000] in <b2b5eee6b3734e9586bbc3ab32e40bc7>:0
  at OpenRA.Mods.Common.Traits.AirstrikePower+<>c__DisplayClass4_0.<SendAirstrike>b__3 (OpenRA.World w) [0x00120] in /home/ivaylo/code/OpenRA-dragunoff/OpenRA.Mods.Common/Traits/SupportPowers/AirstrikePower.cs:195
  at OpenRA.World.Tick () [0x0019e] in /home/ivaylo/code/OpenRA-dragunoff/OpenRA.Game/World.cs:472
  at OpenRA.Game.InnerLogicTick (OpenRA.Network.OrderManager orderManager) [0x001bc] in /home/ivaylo/code/OpenRA-dragunoff/OpenRA.Game/Game.cs:602
  at OpenRA.Game.LogicTick () [0x0003e] in /home/ivaylo/code/OpenRA-dragunoff/OpenRA.Game/Game.cs:629
  at OpenRA.Game.Loop () [0x000f1] in /home/ivaylo/code/OpenRA-dragunoff/OpenRA.Game/Game.cs:793
  at OpenRA.Game.Run () [0x0003c] in /home/ivaylo/code/OpenRA-dragunoff/OpenRA.Game/Game.cs:834
  at OpenRA.Game.InitializeAndRun (System.String[] args) [0x00010] in /home/ivaylo/code/OpenRA-dragunoff/OpenRA.Game/Game.cs:262
  at OpenRA.Program.Main (System.String[] args) [0x00014] in /home/ivaylo/code/OpenRA-dragunoff/OpenRA.Game/Support/Program.cs:31
AL lib: ReleaseALC: 1 device not closed
```


## Additional information
To reproduce order an air-strike near the map edge but coming from the same edge.

--- a/OpenRA.Mods.Common/Traits/Air/AttackBomber.cs
+++ b/OpenRA.Mods.Common/Traits/Air/AttackBomber.cs
@@ -50,27 +50,30 @@ namespace OpenRA.Mods.Common.Traits
 
 		void ITick.Tick(Actor self)
 		{
-			var dat = self.World.Map.DistanceAboveTerrain(target.CenterPosition);
-			target = Target.FromPos(target.CenterPosition - new WVec(WDist.Zero, WDist.Zero, dat));
 			var wasInAttackRange = inAttackRange;
-			var wasFacingTarget = facingTarget;
-
 			inAttackRange = false;
 
-			facingTarget = TargetInFiringArc(self, target, 4 * info.FacingTolerance);
-
-			foreach (var a in Armaments)
+			if (self.IsInWorld)
 			{
-				if (!target.IsInRange(self.CenterPosition, a.MaxRange()))
-					continue;
+				var dat = self.World.Map.DistanceAboveTerrain(target.CenterPosition);
+				target = Target.FromPos(target.CenterPosition - new WVec(WDist.Zero, WDist.Zero, dat));
 
-				inAttackRange = true;
-				a.CheckFire(self, facing, target);
+				var wasFacingTarget = facingTarget;
+				facingTarget = TargetInFiringArc(self, target, 4 * info.FacingTolerance);
+
+				foreach (var a in Armaments)
+				{
+					if (!target.IsInRange(self.CenterPosition, a.MaxRange()))
+						continue;
+
+					inAttackRange = true;
+					a.CheckFire(self, facing, target);
+				}
+
+				// Actors without armaments may want to trigger an action when it passes the target
+				if (!Armaments.Any())
+					inAttackRange = !wasInAttackRange && !facingTarget && wasFacingTarget;
 			}
-
-			// Actors without armaments may want to trigger an action when it passes the target
-			if (!Armaments.Any())
-				inAttackRange = !wasInAttackRange && !facingTarget && wasFacingTarget;
 
 			if (inAttackRange && !wasInAttackRange)
 				OnEnteredAttackRange(self);

--- a/OpenRA.Mods.Common/Traits/SupportPowers/AirstrikePower.cs
+++ b/OpenRA.Mods.Common/Traits/SupportPowers/AirstrikePower.cs
@@ -162,6 +162,7 @@ namespace OpenRA.Mods.Common.Traits
 				});
 
 				aircraft.Add(a);
+				aircraftInRange.Add(a, false);
 
 				var attack = a.Trait<AttackBomber>();
 				attack.SetTarget(self.World, target + targetOffset);
@@ -192,7 +193,6 @@ namespace OpenRA.Mods.Common.Traits
 					a.QueueActivity(new Fly(a, Target.FromPos(target + spawnOffset)));
 					a.QueueActivity(new Fly(a, Target.FromPos(finishEdge + spawnOffset)));
 					a.QueueActivity(new RemoveSelf());
-					aircraftInRange.Add(a, false);
 					distanceTestActor = a;
 				}
 

